### PR TITLE
[SR-63] feat(frontend): add Themes page and components

### PR DESCRIPTION
**Pull request by Sergei-S-V**
# Task

Created a frontend page and components for Themes. 
Extracted reusable logic and UI for the modal component. 
Added tests.

[Link to Ticket](https://incredihire-academy.atlassian.net/jira/software/c/projects/SR/boards/3?selectedIssue=SR-63)

## Acceptance Criteria
- [ ] N/A

### CODE REVIEW

**Review by mcverter — COMMENTED** [review]
This is really great work, @Sergei-S-V .  Thanks for doing it.  We've been wanting to implement a generic FormModal for a while now.  I am learning a lot from your usage of Generics in the code below.  I think either we need to ask @Eugene-Polonsky to relax codacy rules for `any` or you should just `eslint-disable-next-line` everywhere codacy argues with you

File: frontend/src/components/Common/FormModal.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+}
+
+export interface FormModalProps<T> {
+  isOpen: boolean
+  onClose: () => void
+  title: string
+  fields: FieldDefinition<T>[]
+  mutationFn: (data: T) => Promise<any>
+  successMessage: string
+  queryKeyToInvalidate?: string[]
```
Comment (mcverter) [inline frontend/src/components/Common/FormModal.tsx:45]: nit: change this variable name to `queryKey`

Comment (Sergei-S-V) [reply frontend/src/components/Common/FormModal.tsx:66]: It indicates which data (identified by query keys for each data fetch request) should be fetched again from the server after a successful change, ensuring the information remains current.

Comment (mcverter) [reply frontend/src/components/Common/FormModal.tsx:66]: It's a minor point.  The string "item" is the string Tanastack Query uses to keep track of transactions.  What do with it in the modal is InvalidateIt.  But the QueryString can be used for other purposes -- it does not exist only for invalidation.

This is an academic argument, though.  Either way is fine.

File: frontend/src/components/Themes/ThemesTable.columns.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+const columnHelper = createColumnHelper<ThemePublic>()
+
+export const themesColumns: ColumnDef<ThemePublic, string>[] = [
+  columnHelper.accessor("name", {
+    header: "Text",
+    cell: (info) => info.getValue(),
+  }),
+  columnHelper.accessor("description", {
+    header: "Created At",
```
Comment (ilhanbae) [inline frontend/src/components/Themes/ThemesTable.columns.ts:12]: Should this header value be "description"?

image

Comment (Sergei-S-V) [reply frontend/src/components/Themes/ThemesTable.columns.ts]: yes, indeed, missed that, corrected

**Review by ilhanbae — APPROVED** [review]
Looks good, everything works! Thank you for creating reusable modal component too.

**Review by mcverter — COMMENTED** [review]
Great work!

**Review by mcverter — APPROVED** [review]

**Pull request merged**